packages/infra/lib/api-stack.ts

```diff
@@ -665,6 +676,51 @@ export class APIStack extends Stack {
     withingsResource.addMethod("ANY", new apig.LambdaIntegration(withingsLambda));
   }
 
+  private setupFitbitWebhookAuth(ownProps: {
+    lambdaLayers: lambda.ILayerVersion[];
+    baseResource: apig.Resource;
+    vpc: ec2.IVpc;
+    fargateService: ecs_patterns.NetworkLoadBalancedFargateService;
+    fitbitClientSecret: string;
+    envType: string;
+    sentryDsn: string | undefined;
+    alarmAction: SnsAction | undefined;
+  }) {
+    const {
+      lambdaLayers,
+      baseResource,
+      vpc,
+      fargateService: server,
+      fitbitClientSecret,
+      envType,
+      sentryDsn,
+      alarmAction,
+    } = ownProps;
+
+    const fitbitLambda = createLambda({
+      stack: this,
+      name: "Fitbit",
+      runtime: lambda.Runtime.NODEJS_18_X,
+      entry: "fitbit",
+      layers: lambdaLayers,
+      envVars: {
+        API_URL: `http://${server.loadBalancer.loadBalancerDnsName}/webhook/fitbit`,
+        ENV_TYPE: envType,
+        FITBIT_CLIENT_SECRET: fitbitClientSecret,
+        ...(sentryDsn ? { SENTRY_DSN: sentryDsn } : {}),
+      },
+      vpc,
+      alarmSnsAction: alarmAction,
+    });
+
+    // Grant lambda access to the api server
+    server.service.connections.allowFrom(fitbitLambda, Port.allTcp());
+
+    // setup $base/fitbit path with token auth
+    const fitbitResource = baseResource.addResource("fitbit");
+    fitbitResource.addMethod("ANY", new apig.LambdaIntegration(fitbitLambda));
```

Comment:
won't this integration only be for POST requests, not ANY?

Reply:
Resolved. Both POST and GET handled.

Reply:
No `OPTIONS`, right?